ark/utils/deepcell_service_utils.py

```diff
@@ -322,14 +322,14 @@ def _convert_deepcell_seg_masks(seg_mask: bytes) -> np.ndarray:
 
     Returns:
         np.ndarray: The segmentation masks, converted from floating point 64-bit to integer
-        16-bit via `scipy.stats.rankdata`
+        32-bit via `scipy.stats.rankdata`
     """
     float_mask = imread(BytesIO(seg_mask))
 
     # Reshape as ranked_mask returns a 1D numpy array, dims:  n^2 x 1 -> 1 x n x n
     shape = float_mask.shape
 
     # Create the ranked mask
-    ranked_mask: np.ndarray = stats.rankdata(float_mask).astype(dtype="int16").reshape(shape)
+    ranked_mask: np.ndarray = stats.rankdata(float_mask).astype(dtype="int32").reshape(shape)
```

Comment:
The `cell_size`$=0.0$ issue happened due an issue with segmentation masks generated by Deepcell. After they go through the rankdata function, and get converted to `int16`, the ranks integer overflowed leading to negative values in the mask. Following this, when regionprops gets called on these negative values it defaults the area to $0.0$.

Reply:
To follow up, some of the underlying unique cell IDs being assigned in the segmentation mask during the call to `_convert_deepcell_seg_masks`, well beyond what an `'int16'` can handle. There are just over 16000 cells in the sample dataset across all 10 FOVs, but some of them are being assigned cell IDs in the millions.

<img width="435" alt="Screen Shot 2022-09-02 at 6 42 13 PM" src="https://user-images.githubusercontent.com/31424707/188251052-731dbb35-8ff7-4ec6-9cc8-a1637e8ebbfd.png">

In any case, the label assignment is extremely non-contiguous for the sample dataset. This isn't really a big issue (unless there's a section of code requiring a contiguous labeling scheme I can't think of right now) but might be worth correcting these labels to something a little more intuitive.

Reply:
Is this still an issue even after the `int32` fix, or that solves the problem with extremely large values for cell_id?

Reply:
@ngreenwald yes it removes the size 0 cells from the dataset. @srivarra once you're done verifying that everything else runs OK, let's deploy a new Docker image with the `datasets` library included.

Reply:
Right, I get that it removes the size 0 cells, but are there still cell IDs in the millions?